docs/DeveloperGuide.md

```diff
@@ -1,12 +1,406 @@
 # Developer Guide
+## Contents
+- [Acknowledgements](#acknowledgements)
+- [Design & Implementation](#design--implementation)
+- [Implementation](#implementation)
+  - [List](#list)
+  - [Add](#add)
+  - [Edit](#edit)
+  - [Remove](#remove)
+  - [Search](#search)
+  - [Filter](#filter)
+  - [Alert](#alert)
+  - [History](#history)
+  - [Category](#category)
+- [Product Scope](#product-scope)
+  - [Target User Profile](#target-user-profile)
+  - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for Manual Testing](#instructions-for-manual-testing)
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The documentation and organisation of our project follows the recommended format as shown in [SE-Education](http://se-education.org/addressbook-level3/DeveloperGuide.html)
 
-## Design & implementation
+## 1. Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1.1. Architecture Design Diagram
+![Architecture Diagram](ArchitectureDiagram.png)
+
+Breakdown of each component and its role in the application:
+
+`MagusStock`: This is the entry point for the application, and it's responsible for starting the application and coordinating the interactions between the other components.
+
+`ParserHandler`: This component is responsible for handling user input and determining which parser to execute based on the input. It uses a Parser to parse the input and generate a corresponding Command object.
+
+`Parser`: This component is responsible for parsing the user input and generating a Command object. The ParserHandler uses the Parser to parse the input and determine which Command object to create.
+
+`SessionManager`: This component is responsible for managing the inventory data persistence. It's connected to the Storage component, which reads and writes inventory data to and from a CSV file.
+
+`Storage`: This component is responsible for reading and writing inventory data to a CSV file. It's connected to the SessionManager, which uses it to manage the inventory data persistence.
+
+`Ui`: This component is responsible for displaying information to the user. It receives messages and data from the other components and displays them to the user.
+
+Overall, the architecture diagram shows how the different components of the MagusStock application work together to provide the user with a command-line interface for managing an inventory of items. The components are designed to be modular and loosely coupled, allowing for easy modification and extension of the application.
+
+### 1.2. UML Sequence Diagram
+
+![Sequence Diagram](SequenceDiagram.png)
```

Comment:
Classes have a missing `:` at the start. For example, `:MagusStock`. 

Activation bar for `MagusStock` is also missing a closing line. 
![image](https://user-images.githubusercontent.com/84615041/228706370-7a9ef577-3f79-4046-953c-c95cdff765da.png)